flake.nix

```diff
@@ -0,0 +1,43 @@
+{
+  description = "stellar-cli development shell";
+
+  inputs = {
+    nixpkgs.url      = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-utils.url  = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+      in
+      with pkgs;
+      {
+        devShells.default = mkShell {
+          buildInputs = [
+            openssl
+            pkg-config
+            libudev-zero
+            jq
+            (rust-bin.stable.latest.default.override {
+              extensions = [ "rust-src" ];
+              targets = [ "wasm32-unknown-unknown" ];
+            })
+          ];
+          shellHook =
+          ''
+            echo "Using `nix --version`"
+            alias stellar="cargo run --bin stellar --"
+            [ -f ./local.sh ] && source ./local.sh
```

Comment:
Executing local.sh seems surprising, and user specific. Can we leave that out? It seems simple enough for folks to run any setup they need.

Fwiw folks can simply put STELLAR_ env vars into their shell file if they want default values to be used.

Reply:
>Fwiw folks can simply put STELLAR_ env vars into their shell file if they want default values to be used.

You mean shell config file (`bashrc`/`zshrc`)?

I originally added this `local.sh` with the idea of developers being able to put something extra in there easily (like importing a private key or exporting extra variables), which is probably not something super useful.

As for env vars (if we are talking about shell config files), the idea is that we want to keep default shell "clean" from anything related to cli development. 
For example, I may want to use my regular cli with a different account or network exported in the env var defined in my .rc file.
In reality, I'm sure most people already just store their env vars there and re-define in the shell when necessary

So I can just remove it before the merge 👍🏼